akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolInterface.scala

```diff
@@ -236,19 +236,19 @@ private[http] object PoolInterface {
   }
 
   /**
-   * LogSource for PoolGateway instances
+   * LogSource for pool instances
    *
-   * Using this LogSource allows us to set the log class to `PoolGateway` and the log source string
+   * Using this LogSource allows us to set the log class to `PoolId` and the log source string
    * to a descriptive name that describes a particular pool instance.
    */
-  private[http] val GatewayLogSource: LogSource[PoolGateway] =
-    new LogSource[PoolGateway] {
-      def genString(gateway: PoolGateway): String = {
-        val scheme = if (gateway.hcps.setup.connectionContext.isSecure) "https" else "http"
-        s"Pool(${gateway.gatewayId.name}->$scheme://${gateway.hcps.host}:${gateway.hcps.port})"
+  private[http] val PoolLogSource: LogSource[PoolId] =
+    new LogSource[PoolId] {
+      def genString(poolId: PoolId): String = {
+        val scheme = if (poolId.hcps.setup.connectionContext.isSecure) "https" else "http"
```

Comment:
It feels a bit weird to 'reach into' the ID like that. Might be neater to have a method on `PoolId` that constructs this string? OTOH this keeps all logging-related logic here, might be fine.

Reply:
I agree, it's a bit weird (but unchanged from before). If there would be better ways of showing that kind of info with default logging configuration (like putting it into the MDC), that would be preferable.